tests/cloner_test.go

```diff
@@ -750,6 +751,104 @@ var _ = Describe("Namespace with quota", func() {
 	})
 })
 
+var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:component]Cloner Test Suite", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+
+	var sourcePvc *v1.PersistentVolumeClaim
+	var targetPvc *v1.PersistentVolumeClaim
+
+	AfterEach(func() {
+		if sourcePvc != nil {
+			By("[AfterEach] Clean up source PVC")
+			err := f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if targetPvc != nil {
+			By("[AfterEach] Clean up target PVC")
+			err := f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("[test_id:3999] Create a data volume importing an image and then clone it and verify retry count", func() {
+		pvcDef := utils.NewPVCDefinition(sourcePVCName, "1G", nil, nil)
+		pvcDef.Namespace = f.Namespace.Name
+		sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile)
+		targetNs, err := f.CreateNamespace(f.NsPrefix, map[string]string{
+			framework.NsPrefixLabel: f.NsPrefix,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		f.AddNamespaceToDelete(targetNs)
+		doFileBasedCloneTest(f, pvcDef, targetNs)
+
+		By("Verify retry annotation on PVC")
+		targetDvName := "target-dv"
+		targetPvc, err := utils.WaitForPVC(f.K8sClient, targetNs.Name, targetDvName)
+		Expect(err).ToNot(HaveOccurred())
+		restartsValue, status, err := utils.WaitForPVCAnnotation(f.K8sClient, targetNs.Name, targetPvc, controller.AnnPodRestarts)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(status).To(BeTrue())
+		Expect(restartsValue).To(Equal("0"))
+
+		By("Verify the number of retries on the datavolume")
+		dv, err := f.CdiClient.CdiV1alpha1().DataVolumes(targetNs.Name).Get(targetDvName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(dv.Status.RestartCount).To(BeNumerically("==", 0))
+	})
+
+	It("[test_id:4000] Create a data volume importing an image and then clone it while killing the container and verify retry count", func() {
+		By("Prepare source PVC")
+		pvcDef := utils.NewPVCDefinition(sourcePVCName, "1G", nil, nil)
+		pvcDef.Namespace = f.Namespace.Name
+		sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile)
+
+		By("Create clone DV")
+		targetNs, err := f.CreateNamespace(f.NsPrefix, map[string]string{
```

Comment:
This is more an FYI than an issue. The targetNS and addNamespaceToDelete are for when you are cloning across namespaces. This particular test we don't care about that and could have used the same namespace and saved us writing some code.